src/zenml/environment.py

```diff
@@ -109,44 +123,262 @@ def in_paperspace_gradient() -> bool:
         """If the current python process is running in Paperspace Gradient."""
         return "PAPERSPACE_NOTEBOOK_REPO_ID" in os.environ
 
-    @classmethod
-    @contextmanager
-    def _layer(
-        cls, step_is_running: Optional[bool] = None
-    ) -> Iterator["Environment"]:
-        """Contextmanager to temporarily set attributes on the singleton
-        instance.
-
-        These attributes will only be persisted for the active duration of this
-        contextmanager:
-        ```python
-        env = Environment()
-        print(env.step_is_running)  # False
-
-        with Environment._layer(step_is_running=True):
-            print(env.step_is_running)  # True
-
-        print(env.step_is_running)  # False
-        ```
-
-        Calls to this contextmanager can also be nested:
-        ```python
-        with Environment._layer(...):
-            # only attributes from outer context manager are set
-            with Environment._layer(...):
-                # attributes from outer and inner context manager are set
-                # (inner context manager can overwrite values from outer one)
-
-            # only attributes from outer context manager are set
-        ```
+    def register_component(
+        self, name: str, component: "BaseEnvironmentComponent"
```

Comment:
I guess we could remove the name parameter here and use `component.NAME` instead? Or should we allow registering components with a different name?

Reply:
yes, good idea